Include/internal/pycore_genobject.h

```diff
@@ -21,7 +34,30 @@ PyGenObject *_PyGen_GetGeneratorFromFrame(_PyInterpreterFrame *frame)
     return (PyGenObject *)(((char *)frame) - offset_in_gen);
 }
 
-PyAPI_FUNC(PyObject *)_PyGen_yf(PyGenObject *);
+// Mark the generator as executing. Returns true if the state was changed,
+// false if it was already executing or finished.
+static inline bool
+_PyGen_SetExecuting(PyGenObject *gen)
```

Comment:
since this function can fail, maybe name it as `_PyGen_TrySetExecuting` like other free-threading functions which can fail like `TryIncref`